docs/src/references/tedge-log-management.md

```diff
@@ -0,0 +1,232 @@
+The `tedge-log-plugin` parses this configuration file on startup for all the `type` values specified,
+and sends the supported log types message to MQTT local broker on `te/device/main///cmd/log_upload` topic as follows:
+
+```json
+{
+  "types" : [ "mosquitto", "software-management", "c8y_CustomOperation" ]
+}
+```
+
+The plugin continuously watches this configuration file for any changes and resends the JSON message with the `type`s in this file,
+whenever it is updated.
+
+:::note
+If the file `/etc/tedge/plugins/tedge-log-plugin.toml` is ill-formed or cannot be read,
+then a JSON message with an empty array for the `types` field is sent, indicating no log files are tracked.
+:::
+
+## Startup
+
+The plugin declares **supported operation** and **supported log types** on startup.
+It's up to cloud mappers, if it converts the declaration messages to the corresponding cloud format or discard them.
+
+### Supported operation
+
+On startup, the plugin declares to the Cumulocity mapper that it supports `c8y_LogfileRequest` operation
+by creating an empty file at `/etc/tedge/operations/c8y/c8y_LogfileRequest`.
+For other cloud platforms, it does nothing as of now.
+
+### Supported log types
+On startup, the plugin also reports all the log file types that it manages, defined in the `tedge-log-plugin.toml`, to the MQTT topic `te/device/main///cmd/log_upload` with JSON payload,
+as defined in the Configuration section.
+
+## Handling log requests from cloud mapper
+
+After all initialization, this plugin subscribes to `te/device/main///cmd/log_upload/+` topic, listening for 
+log file upload request from any cloud mapper, like this one.
+
+Topic (provided "1234" is operation ID):
+```text
+te/device/main///cmd/log_upload/1234
+```
+
+Payload:
+```json
+{
+  "status": "init",
+  "uploadUrl": "http://127.0.0.1:8000/tedge/file-transfer/main/log_upload/mosquitto",
+  "logType": "mosquitto",
+  "dateFrom": "2013-06-22T17:03:14.000+02:00",
+  "dateTo": "2013-06-22T18:03:14.000+02:00",
+  "searchText": "ERROR",
+  "maximumLines": 1000
+}
+```
+
+The plugin then checks the `tedge-log-plugin.toml` file for the log type in the incoming message (`mosquitto`),
+retrieves the log files using the `target` glob pattern provided in the plugin config file,
+including only the ones modified within the date range(`2013-06-22T17:03:14.000+02:00` to `2013-06-22T18:03:14.000+02:00`),
+with the content filtered by the search text(`ERROR`) and the maximum line count(`1000`).
+
+This filtered content is then uploaded to the given url as `uploadUrl` via HTTP PUT request.
+
+During this process, the plugin need to update the operation status via MQTT for `executing `, `uploading` and `failed` 
+by publishing a message to the topic `te/device/main///cmd/log_upload/<op_id>`.
+
+So, for example, the operation status update message should look like below. The topic is `te/device/main///cmd/log_upload/1234`.
+
+```json
+// Why do we need to repeat other fileds than status and reason?
```

Comment:
I don't think we need all of that, especially since we're sending the responses to a specific `op_id` topic. Just the `status` and `uploadUrl` are more than enough on success. Just for more context, we can add just the `logType` as well, if needed, but nothing else.

Reply:
Yeah, your comment fits what I thought. I will update.

Reply:
I decided to propose to include all information for a command inside the payload for the robustness in the future. If the component who is supposed to receive the message is not interested in other fields, they can just ignore. On the contrary, it's more difficult to get missing data.